apps/finance/app/src/components/useFilteredTransfers.js

```diff
@@ -0,0 +1,106 @@
+import { useState, useCallback, useMemo } from 'react'
+import { endOfDay, isWithinInterval, startOfDay } from 'date-fns'
+import { TRANSFER_TYPES, Incoming, Outgoing } from '../transfer-types'
+import { addressesEqual } from '../lib/web3-utils'
+
+const UNSELECTED_TOKEN_FILTER = -1
+const UNSELECTED_TRANSFER_TYPE_FILTER = -1
+const UNSELECTED_DATE_RANGE_FILTER = { start: null, end: null }
+
+function useFilteredTransfers({ transactions, tokens }) {
+  const [page, setPage] = useState(0)
+  const [selectedDateRange, setSelectedDateRange] = useState(
+    UNSELECTED_DATE_RANGE_FILTER
+  )
+  const [selectedTransferType, setSelectedTransferType] = useState(
+    UNSELECTED_TRANSFER_TYPE_FILTER
+  )
+  const [selectedToken, setSelectedToken] = useState(UNSELECTED_TOKEN_FILTER)
+  const handleSelectedDateRangeChange = useCallback(range => {
+    setPage(0)
+    setSelectedDateRange(range)
+  }, [])
+  const handleTokenChange = useCallback(index => {
+    setPage(0)
+    setSelectedToken(index || UNSELECTED_TOKEN_FILTER)
+  }, [])
+  const handleTransferTypeChange = useCallback(index => {
+    setPage(0)
+    setSelectedTransferType(index || UNSELECTED_TRANSFER_TYPE_FILTER)
+  }, [])
+  const handleClearFilters = useCallback(() => {
+    setPage(0)
```

Comment:
Rather than repeating `setPage()` here, it would likely be nicer to run as a `useEffect()` instead—any time the filters change, we reset the page back to 0.

(Assuming we keep `page` here)

Reply:
Let's see how this looks inside the hook!